libs/pika/async_cuda/tests/unit/cuda_sender.cu

```diff
@@ -126,8 +125,16 @@ int pika_main(pika::program_options::variables_map& vm)
     std::size_t device = vm["device"].as<std::size_t>();
     //
     unsigned int seed = (unsigned int) std::time(nullptr);
+#if !defined(PIKA_HAVE_HIP)
+    // We undefined this due to a internal compiler error with ROCm Clang-15
+    // triggered when calling the count() method on the variables_map.
```

Comment:
With appropriate line-wrapping (I'm unsure if the suggestion is longer than 80 characters):
```suggestion
    // ROCm Clang-15 (HIP 5.3.3) fails to compile this with an internal compiler error.
    // See https://github.com/pika-org/pika/issues/585 for more details.
#if !defined(PIKA_HAVE_HIP)
```